### PR TITLE
Testing React Components

### PR DESCRIPTION
**Pull request by Alaska01**
This project is a basic react calculator app. Here, I used React components, JSX, React states and props, and event handlers. Testing was done with the following dependencies, Enzyme a JavaScript Testing utility for React  and React renderer that can be used to render React components as pure JavaScript objects, without depending on the DOM.

### CODE REVIEW

File: src/__test__/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  const result = operate(5, 8, '+');
+  expect(result).not.toBe('10');
+});
+
+it('returns the subtraction of two numbers', () => {
+  const result = operate(8, 5, '-');
+  expect(result).toBe('3');
+});
+
+it('returns the subtraction of two numbers', () => {
```
Comment (afanimartin) [inline src/__test__/calculate.test.js]: - Am guessing this is the negative test case but you are describing it just like the positive one. This is inconsistent with the first negative test. I expect it to be `returns the subtraction of two numbers as not equal`. Please make the required changes for all test cases following this one.

File: src/__test__/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+it('returns the division of two digits', () => {
+  const result = calculate({
+    total: '10', next: '5', operation: '÷', totalCalc: true,
+  }, '=');
+  expect(result).toStrictEqual({
+    total: '2', next: null, operation: null, totalCalc: false,
+  });
+});
+
+it('returns the division of two digits', () => {
```
Comment (afanimartin) [inline src/__test__/calculate.test.js]: - Please add test cases for the operators: `AC`, `+/-` and `%` as it is a good idea to fully cover all your logic with tests.

**Review by afanimartin — CHANGES_REQUESTED** [review]
Hi @Alaska01, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

### Martin Afani
#### Slack: whiz25

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__test__/calculate.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const result = calculate({
+    total: '10', next: '5', operation: '÷', totalCalc: true,
+  }, '=');
+  expect(result).not.toStrictEqual({
+    total: '2', next: null, operation: null, totalCalc: true,
+  });
+});
```
Comment (rukundoeric) [inline src/__test__/calculate.test.js]: - [ ] <b>REQUIRED:</b> As request by the previous reviewer, you have to add test cases for the following operations:(`AC`, `+/-` And `%`) in order to comply with the project requirements.

**Review by rukundoeric — CHANGES_REQUESTED** [review]
Hi @Alaska01, 

_Great job so far 👏 💯 👍. You've done well implementing some of the requested changes. However, there are still some which aren't addressed yet. Please, check the comments below._
## Highlights
  - Follows git workflow.  💯
  - Codebase looks clean and professional. 💯 
  - Addressed some of the issues raised from the previous review. 💯 

## Required Changes ♻️
  - [ ] Check the comments under the review.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
  
  - N/A 

Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by rukundoeric — APPROVED** [review]
Hi @Alaska01,

_Excellent job on this project. Keep it up 🎆_

_Congratulations! 🎉  on the splendid job 👍
You can now merge this PR and submit the project completion form 🚀 . I wish you all the very best in your next tasks 🥇_


Cheers and Happy coding!👏👏👏

__Feel free to leave any questions or comments in the PR thread if something is not 100% clear.__

------
>As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**